xUnique.py

```diff
@@ -260,14 +263,13 @@ def file_dir_cmp(x, y):
             if children_match:
                 output_u8line(line)
                 child_flag = True
-                if isinstance(fc_end_ptn, unicode):
+                if isinstance(fc_end_ptn, six.text_type):
                     fc_end_ptn = re_compile(children_match.group(1) + fc_end_ptn)
             if child_flag:
                 if fc_end_ptn.search(line):
                     if lines:
                         if self.main_group_hex not in last_two[0]:
-                            lines.sort(key=lambda file_str: children_pbx_key_ptn.search(file_str).group(),
-                                       cmp=file_dir_cmp)
+                            lines.sort(key=lambda file_str: (children_pbx_key_ptn.search(file_str).group(), file_dir_order))
```

Comment:
Yep, I know that python3 removed `cmp` parameter in `sort`. But I think your lambda seems not correct. Though I am not very familiar with `cmp` and `key` in `sort` method
But the result could tell the difference. You could run your version and master on the same project file, you would see differences in the sorting dirs and files.